modules/xsession.nix

```diff
@@ -36,75 +65,106 @@ in
     };
   };
 
-  config = mkIf cfg.enable {
-    systemd.user.services.setxkbmap = {
-      Unit = {
-        Description = "Set up keyboard in X";
-        After = [ "graphical-session-pre.target" ];
-        PartOf = [ "graphical-session.target" ];
-      };
-
-      Install = {
-        WantedBy = [ "graphical-session.target" ];
-      };
+  config = mkIf cfg.enable (mkMerge [
+    (mkIf cfg.windowManager.usesDeprecated {
+      warnings = [
+        ("xsession.windowManager is deprecated, "
+          + "please use xsession.windowManager.command")
+      ];
+    })
+
+    (mkIf (cfg.windowManager.xmonad.enable
```

Comment:
So now window manager's options definition and application is separated from each other? Was there some reason on why not to include this part inside the `xmonad` module?

Reply:
Yeah, unfortunately this has to be separate until `xsession.windowManager` as a string is removed and xmonad (and and imminent other window manager module) is a real module. Right now it is interpreted as a submodule and therefore cannot access or manipulate the overall configuration.

Reply:
So you mean that it will be a temporary solution and we will fix it later on, after disabling `xsession.windowManager` option completely?

Reply:
Yes, exactly. Once that happens the `xsession.nix` file should contain nothing related to specific window managers. I'm thinking a month or so should be enough for people to switch away from using `xsession.windowManager` as a string.